logger/syslogd/syslogd.go

```diff
@@ -97,7 +101,7 @@ func (h *handler) mainLoop() {
 }
 
 // Listen starts a new syslog server which runs until it receives a signal.
-func Listen(signalChan chan os.Signal, cleanupDone chan bool, bindAddr string) {
+func Listen(exitChan, cleanupDone chan bool, drainChan chan string, bindAddr string) {
```

Comment:
> If I do, then I'd have to create two separate signal notifiers for two separate channels. As I need to know in main.go when a signal has fired to shutdown the etcd watcher.

The etcd watcher and the log server are two separate events, and we may include infrastructure in the future to handle different signals. I'd rather keep it the way things are if we can.